### ISSUE TITLE
Docker production build from source does not start up, geodb dependency fails

### ISSUE THREAD

**Issue by immanuelfodor** [mozilla/send#1430]
It seems the Send repository Docker build produces a non-working container image because of the same Maxmind GeoIP access change that is added here: https://github.com/mozilla/fxa/issues/3773

There are two error logs in the Docker build output:

```
Step 7/24 : RUN set -x     && npm ci     && npm run build
 ---> Running in a676291f855a
+ npm ci

> fxa-geodb@1.0.4 postinstall /app/node_modules/fxa-geodb
> node scripts/start-db-download.js

{"Timestamp":1577864899815000000,"Logger":"fxa-geodb","Type":"Download directory is /app/node_modules/fxa-geodb/db","Severity":6,"Pid":171,"EnvVersion":"2.0"}
{"Timestamp":1577864899816000000,"Logger":"fxa-geodb","Type":"Adding https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz","Severity":6,"Pid":171,"EnvVersion":"2.0"}
{"Timestamp":1577864899816000000,"Logger":"fxa-geodb","Type":"Setting /app/node_modules/fxa-geodb/db/cities-db.mmdb as target file","Severity":6,"Pid":171,"EnvVersion":"2.0"}
{"Timestamp":1577864899998000000,"Logger":"fxa-geodb","Type":"uncaughtException","Severity":0,"Pid":171,"EnvVersion":"2.0","Fields":{"error":"Error: getaddrinfo ENOTFOUND geolite.maxmind.com geolite.maxmind.com:443","stack":"\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:56:26)"}}
```

Later, another one:

```
Step 19/24 : RUN npm ci --production && npm cache clean --force
 ---> Running in 4d34f435d45b

> fxa-geodb@1.0.4 postinstall /app/node_modules/fxa-geodb
> node scripts/start-db-download.js

{"Timestamp":1577864963654000000,"Logger":"fxa-geodb","Type":"Download directory is /app/node_modules/fxa-geodb/db","Severity":6,"Pid":52,"EnvVersion":"2.0"}
{"Timestamp":1577864963656000000,"Logger":"fxa-geodb","Type":"Adding https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz","Severity":6,"Pid":52,"EnvVersion":"2.0"}
{"Timestamp":1577864963656000000,"Logger":"fxa-geodb","Type":"Setting /app/node_modules/fxa-geodb/db/cities-db.mmdb as target file","Severity":6,"Pid":52,"EnvVersion":"2.0"}
{"Timestamp":1577864963677000000,"Logger":"fxa-geodb","Type":"uncaughtException","Severity":0,"Pid":52,"EnvVersion":"2.0","Fields":{"error":"Error: getaddrinfo ENOTFOUND geolite.maxmind.com geolite.maxmind.com:443","stack":"\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:56:26)"}}
added 361 packages in 4.772s
```

The build is successful, the container starts, but the logs show an error, and the service is not available on the `PORT` specified in the config (env):

```
send          | {"Timestamp":1577862561126000000,"Logger":"FirefoxSend","Type":"uncaughtException","Severity":0,"Pid":1,"EnvVersion":"2.0","Fields":{"error":"Error: ENOENT: no such file or directory, open '/app/node_modules/fxa-geodb/db/cities-db.mmdb'","stack":"\n    at Object.openSync (fs.js:443:3)\n    at Object.readFileSync (fs.js:343:35)\n    at Object.exports.openSync (/app/node_modules/maxmind/index.js:50:30)\n    at module.exports (/app/node_modules/fxa-geodb/lib/fxa-geodb.js:16:28)\n    at Object.<anonymous> (/app/server/amplitude.js:2:35)\n    at Module._compile (internal/modules/cjs/loader.js:778:30)\n    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)\n    at Module.load (internal/modules/cjs/loader.js:653:32)\n    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)\n    at Function.Module._load (internal/modules/cjs/loader.js:585:3)"}}
```

Although the geoip invocation is in a try-catch statement itself, the module load is not, so the app crashes: https://github.com/mozilla/send/blob/master/server/amplitude.js#L25-L29

The quickest fix is to comment out this line, then rebuild the image, it works fine then:
https://github.com/mozilla/send/blob/master/server/amplitude.js#L2

```javascript
// const geoip = require('fxa-geodb')();
```

**Comment by immanuelfodor** [comment]
Just for reference, you can apply this patch to quick fix the problem:

```bash
cat <<EOF > geodb.patch
diff --git a/server/amplitude.js b/server/amplitude.js
index bb1ebcec..a6e1812f 100644
--- a/server/amplitude.js
+++ b/server/amplitude.js
@@ -1,5 +1,5 @@
 const crypto = require('crypto');
-const geoip = require('fxa-geodb')();
+// const geoip = require('fxa-geodb')();
 const fetch = require('node-fetch');
 const config = require('./config');
 const pkg = require('../package.json');
EOF

git apply --stat geodb.patch
git apply --check geodb.patch
git apply geodb.patch

docker-compose up -d --build
docker-compose logs -f
```

It still needs attention from the team to provide a long-term solution. Maybe, if the [`AMPLITUDE_ID`](https://github.com/mozilla/send/blob/master/server/config.js#L83-L87) is not set, then the whole file can be skipped.

**Comment by dannycoates** [comment]
fixed here: 8ad45971145dd27089031296769982e1ea5ce426
